### PR TITLE
Revise waypoint reached

### PR DESCRIPTION
**Pull request by patrick-5546**
Addresses #171.

### CODE REVIEW

File: python/Path.py [hunk 1]
```diff
@@ -430,6 +438,19 @@ def x(y):
...
+        nextWaypointLatlon = self._latlons[self._nextWaypointIndex]
+
+        return self.waypointReached(positionLatlon, previousWaypointLatlon, nextWaypointLatlon)
+
+    # assumes there are at least 2 elements in self._latlons
+    def lastWaypointReached(self, positionLatlon):
+        previousWaypointLatlon = self._latlons[len(self._latlons) - 2]
+        nextWaypointLatlon = self._latlons[len(self._latlons) - 1]
+
+        return self.waypointReached(positionLatlon, previousWaypointLatlon, nextWaypointLatlon)
```
Comment (patrick-5546) [inline python/Path.py:452]: Issues with this implementation:
* Returns true every time the simulation starts
* Assumes that there are at least 2 elements in `self._latlons`

**Review by tylerlum — COMMENTED** [review]
Looks good to me so far. Update the unit tests to ensure proper functionality. I will do some testing myself and approve after :)

**Review by tylerlum — CHANGES_REQUESTED** [review]
Great work. Make sure to add in those unit tests to gain confidence in the code, and double check the verticalShift math.

**Review by tylerlum — APPROVED** [review]
Works beautifully. Might still be a good idea to draw the boundary line in another PR. Go ahead and merge.

**Pull request merged**